plugins/s3/src/plugin.cpp

```diff
@@ -0,0 +1,307 @@
+                                         output_stream.status().ToString()));
+    }
+    auto stream_guard
+      = caf::detail::make_scope_guard([this, &ctrl, output_stream]() {
+          auto status = output_stream.ValueUnsafe()->Close();
+          if (not output_stream.ok()) {
+            ctrl.abort(
+              caf::make_error(ec::filesystem_error,
+                              fmt::format("failed to close output stream for "
+                                          "URI "
+                                          "`{}`: {}",
+                                          args_.uri.inner, status.ToString())));
+          }
+        });
+    return [&ctrl, output_stream, uri = args_.uri.inner,
+            guard = std::make_shared<decltype(s3_guard)>(std::move(s3_guard)),
+            stream_guard = std::make_shared<decltype(stream_guard)>(
+              std::move(stream_guard))](chunk_ptr chunk) mutable {
+      if (!chunk || chunk->size() == 0)
+        return;
+      auto status
+        = output_stream.ValueUnsafe()->Write(chunk->data(), chunk->size());
+      if (not output_stream.ok()) {
+        ctrl.abort(caf::make_error(ec::filesystem_error,
+                                   fmt::format("failed to write to output "
+                                               "stream for URI "
+                                               "`{}`: {}",
+                                               uri, status.ToString())));
+        return;
+      }
+    };
+  }
+
+  auto name() const -> std::string override {
+    return "s3";
+  }
+
+  auto default_printer() const -> std::string override {
+    return "json";
+  }
+
+  auto is_joining() const -> bool override {
+    return true;
+  }
+
+  friend auto inspect(auto& f, s3_saver& x) -> bool {
+    return f.object(x).pretty_name("s3_saver").fields(f.field("args", x.args_));
+  }
+
+private:
+  s3_args args_;
+};
+
+class plugin final : public virtual loader_plugin<s3_loader>,
+                     public virtual saver_plugin<s3_saver> {
+public:
+  auto initialize([[maybe_unused]] const record& plugin_config,
+                  [[maybe_unused]] const record& global_config)
+    -> caf::error override {
+    return caf::none;
+  }
+
+  auto parse_loader(parser_interface& p) const
+    -> std::unique_ptr<plugin_loader> override {
+    auto parser = argument_parser{
+      name(),
+      fmt::format("https://docs.tenzir.com/docs/next/connectors/{}", name())};
+    auto args = s3_args{};
+    parser.add("-x,--anonymous", args.anonymous);
+    parser.add(args.uri, "<uri>");
+    parser.parse(p);
+    return std::make_unique<s3_loader>(std::move(args));
+  }
+
+  auto parse_saver(parser_interface& p) const
+    -> std::unique_ptr<plugin_saver> override {
+    auto parser = argument_parser{
+      name(),
+      fmt::format("https://docs.tenzir.com/docs/next/connectors/{}", name())};
+    auto args = s3_args{};
+    parser.add("-x,--anonymous", args.anonymous);
```

Comment:
I talked about the `-x` short option with Rémi and Jannis at the retreat sometime late in the night, and we considered it to be confusing. Our vote was to remove the short option entirely, and to only support the spelled-out `--anonymous`.

Reply:
I removed `-x`.